app/src/test/webapp/app/error/error.ctrl.test.js

```diff
@@ -178,4 +190,47 @@ describe("ErrorCtrl", function() {
         });
     });
   });
+
+  describe("after calling resubmit", function() {
+    it("should call resubmitMessage() in service with a resubmittable message", function() {
+      spyOn(msgSvc, 'resubmitMessage').and.callFake(function() {
+        return $q.when({
+          data: {
+            status: "Success"
+          }
+        })
+      });
+
+      $controller("ErrorCtrl", {
+        $scope: $scope
+      });
+
+      $scope.message = resubmittableMessage;
+
+      $scope.resubmitMessage();
+      $scope.$apply();
```

Comment:
I believe $scope.apply() is not necessary here, we don't care about the promise resolving.